include/sas.h

```diff
@@ -304,6 +304,10 @@ class SAS
                                     const char *fmt,
                                     ...);
 
+  typedef int (socket_factory_t)(const char* hostname,
+                                 const char* port);
```

Comment:
I think it would also be useful to comment this typedef to explain what the function is for.